### ISSUE TITLE
Confusing error message on loading and checking SequenceControl information.

### ISSUE THREAD

**Issue by lazykyama** [NVIDIA/tensorrt-inference-server#842]
**Is your feature request related to a problem? Please describe.**

I was quite confused by an error message when I tried to use sequence batcher and TF savedmodel. TRTIS said my model dimension of START tensor doesn't match an expected dimension like below.

```
E1108 00:49:45.410652 1 model_repository_manager.cc:813] failed to load 'stream_rnn' version 1: Invalid argument: unable to load model 'stream_rnn', sequence control 'START' dims [1] don't match expected dims [1]
```

However, the root cause of this error is related to an issue that the configuration includes `max_batch_size: 1` although my model doesn't support dynamic batching. This error message is confusing...

**Describe the solution you'd like**

Separating functionalities of `CompareDimsExact()` into appropriate units.

A few lines of `CompareDimsExact()` check if a given model supports dynamic batching (ie, first dimension is -1) or not.
https://github.com/NVIDIA/tensorrt-inference-server/blob/r19.09/src/backends/tensorflow/tf_utils.cc#L38-L42
After these lines, this function checks rank and dimension values.

So, this function can be divided into 2 parts, checking support of dynamic batching and checking rank and dimension values.

**Describe alternatives you've considered**

It's ok to change error message in caller (eg, https://github.com/NVIDIA/tensorrt-inference-server/blob/r19.09/src/backends/tensorflow/savedmodel_backend.cc#L233).

**Additional context**

My environment info are below.

## TRTIS start command

```
nvidia-docker run -it --rm --shm-size=1g --ulimit memlock=-1 --ulimit stack=67108864 \
    -p8000:8000 -p8001:8001 -p8002:8002 -v $(pwd)/models/:/tmp/models \
    nvcr.io/nvidia/tensorrtserver:19.09-py3 /opt/tensorrtserver/bin/trtserver \
    --model-store=/tmp/models
```

## Configuration

```
name: "stream_rnn"
platform: "tensorflow_savedmodel"
max_batch_size: 1
sequence_batching {
    max_sequence_idle_microseconds: 5000000
    control_input [
        {
            name: "START"
            control: [
                {
                  kind: CONTROL_SEQUENCE_START
                  fp32_false_true: [ 0, 1 ]
                }
            ]
        },
        {
            name: "READY"
            control: [
                {
                  kind: CONTROL_SEQUENCE_READY
                  fp32_false_true: [ 0, 1 ]
                }
            ]
        }
    ]
}
input [
    {
        name: "INPUT"
        data_type: TYPE_FP32
        dims: [ 1, 128 ]
    }
]
output [
    {
        name: "OUTPUT"
        data_type: TYPE_FP32
        dims: [ 1, 29 ]
    }
]
default_model_filename: "savedmodel"
instance_group [
    {
        count: 1
        kind: KIND_GPU
    }
]
```

## Model information

```
saved_model_cli show --dir /ws/deploy/models/stream_rnn/1/savedmodel/ --tag_set serve --signa
ture_def serving_default
2019-11-08 04:13:34.758001: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.1
The given SavedModel SignatureDef contains the following input(s):
  inputs['INPUT'] tensor_info:
      dtype: DT_FLOAT
      shape: (1, 128)
      name: inference/INPUT:0
  inputs['READY'] tensor_info:
      dtype: DT_FLOAT
      shape: (1)
      name: inference/READY:0
  inputs['START'] tensor_info:
      dtype: DT_FLOAT
      shape: (1)
      name: inference/START:0
The given SavedModel SignatureDef contains the following output(s):
  outputs['OUTPUT'] tensor_info:
      dtype: DT_FLOAT
      shape: (1, 29)
      name: inference/OUTPUT:0
Method name is: tensorflow/serving/predict
```

**Comment by deadeyegoodwin** [comment]
Thanks for the report. We'll look into improving that error message.

**Comment by CoderHam** [comment]
Fixed by #955
